### ISSUE TITLE
Bothering mass indentation change on single line change

### ISSUE THREAD

**Issue by victornoel** [prettier/prettier#2766]
**Prettier Version:** 1.6.1

**Code**

![image](https://user-images.githubusercontent.com/160975/30153585-6e3c28e4-93b7-11e7-8184-4805e6d5b127.png)

Basically, changing one line make all the following indented code be shifted. In this example this is small, but I have huge piece of code that are impacted the same.

This is not a bug per se, I understand why the indentation algorithm produces this change, but it is not optimal I find in terms of code revision to have such impact of one line change.
I'm wondering if there isn't a way to not have the indentation of the different parts of such a pattern inter-dependent like this…

**Comment by azz** [comment]
It's unfortunate but there's probably not a lot we can do. Any suggestions?

**Comment by victornoel** [comment]
Yep, I'm not clear on that either, to detail a bit what I was saying in the description, consider something like this:
```ts
fun( // a
  (params...) => { // b
    body // c
  } // b
); // a

I suppose any "solution" for this situation would imply that the way `b` and `c` are indented should not be dependent.

Example of a result:
```ts
fun( // a
  (params...) => { // b
  body // c
  } // b
); // a

fun((params...) => { // a and b
  body // c
}); // a
```

Or another one:
```ts
fun( // a
  (params...) => { // b
    body // c
  } // b
); // a

fun((params...) => { // a and b
    body // c
}); // a
```

That's not exceptional… :)

**Comment by josephfrazier** [comment]
I'm not sure there's a good way to avoid this, other than by *always* putting function arguments on new lines and indenting them, which probably would not be considered "prettier" by most people...

One workaround you can use is to pass `--ignore-all-space` (`-w` for short) to `git show` or `git diff`. This should make the diff much easier to read.

**Comment by victornoel** [comment]
I agree with your analysis yes, though, I had some more thought about this :)

The problem is really about the opening `{` and not especially the function arguments.

For example this always work:
```ts
fun((params...) =>//a
  { // b
    body // c
  } // b
); // a

// here, there is no need to indent the { (b)
fun(
  (params...) =>// a
  { // b 
    body // c
  } // b
); // a

// if a + b + c can fit on one line, then you can have:
fun((params...) =>{ body }); // a + b + c

// if only b+c fit on one line, then you can have:
fun((params...) =>// a
  { body } // b + c
); // a

// or
fun(
  (params...) =>// a
  { body } // b + c
); // a
```

Or depending on what works best, you can put the `=>` with the `{` when it makes sense, for example:
```ts
fun(
  (params...) // a
  => { // b 
    body // c
  } // b
); // a

fun((params...) // a
  => { body } // b + c
); // a
```

(but I don't think it is "prettier" :)

**Comment by sompylasar** [comment]
Maybe this could help: https://github.com/josephfrazier/prettier-diff

**Comment by kalinchernev** [comment]
I just had a similar question in mind: is it possible to somehow scope the prettier's execution to a specific lines of code instead of a file? The lastly shared project seems interesting working with diffs, but doesn't quite solve for what i'd be dreaming of having.

My scenario: I want to use prettier everywhere i can, all the time, so my editor is configured for it by default.
However, I have to work sometimes with code bases which do not want prettier. So I tick the Require Config from VS Code and leave the settings page open for switching it back off when i'm ready.

I wish prettier could "see" only the git diff, i.e. my current changes, and reformat them, not the whole file.

**Comment by kalinchernev** [comment]
I found a way, involves using VS Code as an editor which can apply formatting on git diff https://twitter.com/kalinchernev/status/1364123712673492994
